src/Service/DownloadService.php

```diff
@@ -97,6 +102,41 @@ public function render(array $data): string
 
     }//end render()
 
+    /**
+     * Downloads a docx.
+     * The html that is added has to be whitout a <head><style></style></head> section.
+     *
+     * @param array $data The data to render for this docx.
+     *
+     * @return string The docx as file output.
+     */
+    public function downloadDocx(array $data): string
+    {
+        $raw = $this->render($data);
+
+        $docx    = new PhpWord();
+        $section = $docx->addSection();
+        Html::addHtml($section, $raw);
+        $file = 'data.docx';
+
+        header("Content-Description: File Transfer");
+        header('Content-Disposition: attachment; filename="'.$file.'"');
+        header('Content-Type: application/vnd.openxmlformats-officedocument.wordprocessingml.document');
+        header('Content-Transfer-Encoding: binary');
+        header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+        header('Expires: 0');
+
+        $docxWriter = IOFactory::createWriter($docx, 'Word2007');
+        $fileId     = Uuid::uuid4();
+        $filename   = '/var/tmp/'.$fileId->toString().'.docx';
+
+        $docxWriter->save($filename);
+        $rendered = \Safe\file_get_contents($filename);
+
```

Comment:
Forgot to tell you, but you can remove the temporary file with unlink($filename);